File: src/main/taparia-model/src/main/java/ru/vityaman/itmo/web/lab/taparia/model/figure/solo/Colored.java
package ru.vityaman.itmo.web.lab.taparia.figure.solo;

import lombok.EqualsAndHashCode;
import lombok.ToString;
import lombok.Value;
import ru.vityaman.itmo.web.lab.taparia.figure.Figure;
import ru.vityaman.itmo.web.lab.taparia.figure.primitive.Point;

@Value
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
public class Colored extends SoloFigure {
    String color;

    public Colored(String color, Figure child) {
        super(child);
        this.color = color;
    }

    @Override
    public boolean contains(Point point) {
        return child().contains(point);
    }
}

File: .build/web/ts/picture/figure/colored.ts
import { Figure } from "./figure.js";

export class Colored extends Figure {
    constructor(
        public readonly color: string,
        public readonly child: Figure
    ) {
        super("colored");
    }

    draw(ctx: CanvasRenderingContext2D): void {
        const pastFillStyle = ctx.fillStyle
        ctx.fillStyle = this.color;
        this.child.draw(ctx);
        ctx.fillStyle = pastFillStyle
    }
}
